### PR TITLE
[YUNIKORN-2581] Expose running placement rules in REST

### PR DESCRIPTION
**Pull request by wilfred-s**
### What is this PR for?
Add a new REST endpoint under the partition that returns the currently active placement rules:
/ws/v1/partition/:partition/placementrules
The call retrieves the active set from the placement manager. This will include the recovery rule. The rule interface is extended with a new private method ruleDAO() that must be implemented by all rules and returns the RuleDAO object.

Add the currently active set of placement rules to the state dump. This handles multiple partitions. It wraps the list of RuleDAO objects in a RuleDAOInfo per partition.

### What type of PR is it?
* [X] - Feature

### Todos
* [ ] - Add documentation

### What is the Jira issue?
* https://issues.apache.org/jira/browse/YUNIKORN-2581

### How should this be tested?
Full unit test provided
e2e: statedump or REST endpoint

### CODE REVIEW

**Comment by wilfred-s** [comment]
The test failure is `TestUpdateNodeCapacity()` [YUNIKORN-2573](https://issues.apache.org/jira/browse/YUNIKORN-2573). We should try and fix that soon to allow clean builds

**Comment by arhtur007** [comment]
> The test failure is `TestUpdateNodeCapacity()` [YUNIKORN-2573](https://issues.apache.org/jira/browse/YUNIKORN-2573). We should try and fix that soon to allow clean builds

Hello Wilfred,
I'll take a look on  this issue this weekend, thank you!

**Review by pbacsko — APPROVED** [review]
+1 LGTM

**Review by chenyulin0719 — APPROVED** [review]
LGTM. Only 1 nit.

File: pkg/scheduler/placement/rule_test.go [hunk 1]
```diff
@@ -52,6 +52,15 @@ func TestNewRule(t *testing.T) {
 	if err != nil || nr == nil {
 		t.Errorf("new normalised newRule build failed which should not, newRule 'nil' , err: %v, ", err)
 	}
+
+	// test normalise newRule name
```
[the remark this answers is not in the document]
Comment (wilfred-s) [reply pkg/scheduler/placement/rule_test.go:56]: fixed on commit

**Pull request closed without merging**